File: src/main/java/org/dedda/bratwurst/parse/PushParser.java
package org.dedda.bratwurst.parse;

import org.dedda.bratwurst.lang.BWExpression;
import org.dedda.bratwurst.lang.Push;

import static org.dedda.bratwurst.parse.Emoji.*;

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
public class PushParser extends InstructionParser {

    public Push parse(String line, int linenumber) {
        if (line.matches(Patterns.PUSH)) {
            String expression = line.substring(LEMON.length(), line.length() - SANTA.length());
            BWExpression argument = new ExpressionParser().parse(expression, linenumber);
            return new Push(linenumber, argument);
        }
        return null;
    }

}

File: src/main/java/org/dedda/bratwurst/parse/PushParser.kt
package org.dedda.bratwurst.parse

import org.dedda.bratwurst.lang.Push

/**
 * Created by dedda on 12/8/15.
 *
 * @author dedda
 */
class PushParser : InstructionParser() {
    override fun parse(line: String, lineNumber: Int): Push? {
        if (line.matches(PATTERN)) {
            val expression = line.substring(1, line.length - 1)
            val argument = ExpressionParser().parse(expression, lineNumber)
            return Push(lineNumber, argument)
        }
        return null
    }

    companion object {
        private val PATTERN = Regex(Patterns.PUSH)
    }
}